activation/post_test.go

```diff
@@ -34,32 +35,42 @@ func TestPostSetupManager(t *testing.T) {
 	mgr, err := NewPostSetupManager(id, cfg, logtest.New(t), cdb, goldenATXID)
 	req.NoError(err)
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var eg errgroup.Group
 	lastStatus := &atypes.PostSetupStatus{}
-	go func() {
-		for status := range mgr.StatusChan() {
-			req.True(status.NumLabelsWritten >= lastStatus.NumLabelsWritten)
-			req.Equal(opts, *status.LastOpts)
-			req.Nil(status.LastError)
-
-			if status.NumLabelsWritten == uint64(opts.NumUnits)*cfg.LabelsPerUnit {
-				// TODO(moshababo): fix the following failure. `status.State` changes to `postSetupStateComplete` only after the channel event was triggered.
-				// req.Equal(postSetupStateComplete, status.State)
-			} else {
-				req.Equal(atypes.PostSetupStateInProgress, status.State)
-			}
+	eg.Go(func() error {
+		timer := time.NewTicker(50 * time.Millisecond)
+		defer timer.Stop()
 
-			lastStatus = status
+		for {
+			select {
+			case <-ctx.Done():
+				return nil
+			case <-timer.C:
+				status := mgr.Status()
+				req.True(status.NumLabelsWritten >= lastStatus.NumLabelsWritten)
```

Comment:
How about using `GreaterOrEqual`? It would give better error messages in case of failure.
```suggestion
				req.GreaterOrEqual(status.NumLabelsWritten, lastStatus.NumLabelsWritten)
```